src/frontend/komponenter/Fagsak/Vedtak/VedtakContainer.tsx

```diff
@@ -72,13 +70,35 @@ const VedtakContainer: React.FC<IProps> = ({ behandling, fagsak }) => {
         behandling.behandlingsårsakstype ===
             Behandlingårsak.REVURDERING_FEILUTBETALT_BELØP_HELT_ELLER_DELVIS_BORTFALT;
 
-    const { hentSjekkLikhetPerioder } = useSjekkLikhetPerioder(behandling.behandlingId);
+    const { erPerioderLike, hentSjekkLikhetPerioder } = useSjekkLikhetPerioder(
+        behandling.behandlingId
+    );
+    const { angreSammenslåingAvPerioder } = useAngreSammenslåingPerioder(behandling.behandlingId);
+    const [erPerioderSammenslått, settErPerioderSammenslått] = useState<boolean>(false);
+
+    const { sammenslåPerioder, feilmelding } = useSammenslåPerioder(behandling.behandlingId);
+
+    const handleKnappTrykk = async () => {
+        const oppdaterErPerioderSammenslått = !erPerioderSammenslått;
+        settErPerioderSammenslått(oppdaterErPerioderSammenslått);
+        if (!oppdaterErPerioderSammenslått) {
+            await angreSammenslåingAvPerioder();
+        } else {
+            await sammenslåPerioder();
+        }
+        hentVedtaksbrevtekster();
+    };
 
     useEffect(() => {
-        hentSjekkLikhetPerioder();
-    }, [hentSjekkLikhetPerioder]);
+        const fetch = async () => {
+            const likhetResult = await hentSjekkLikhetPerioder();
+            settErPerioderSammenslått(likhetResult);
```

Comment:
Burde vi ikke hente `skal_sammenslaa_perioder` verdien fra backend her?

Reply:
Vi trenger sjekken for å avgjøre om knappen skal vises. Vi må sjekke på `hentSjekkLikhetPerioder` hvis vi `angrer sammenslåing` og deretter ønsker å `slå sammen` igjen.

Reply:
Okay, men da lurer jeg på dette scenariet. 

Jeg kommer til brevsiden og "angrer sammenslåing". 
Jeg lukker behandlingen og åpner den på nytt senere. 
Da vil jo "erPerioderLike" være true, men jeg har valgt å splitte periodene allikevel.

Mulig det er noe jeg ikke forstår, men slik jeg leser flyten så blir det i frontend seende ut som vi har slått sammen periodene, mens det i backend ikke er gjort dette.

Reply:
Du har rett - som vanlig!😄 Har laget et endepunkt i backend for å hente verdien, denne blir brukt for å sette korrekt state og vise riktig tekst på knappen.